Using not reserved words like "link" and etc throws an C error

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**v version**:
V 0.2.4 227d12f

**OS:**  ArchLinux
```
uname -a
Linux supervoid 5.13.13-arch1-1 #1 SMP PREEMPT Thu, 26 Aug 2021 19:14:36 +0000 x86_64 GNU/Linux
```

Please get the code from:
https://github.com/ep4sh/algoV/blob/master/linked_list_buggy.v

**What did you do?**
```
v run linked_list_buugy.v
```
**What did you expect to see?**
*no output* (assert is working as expected)
 
**What did you see instead?**
```
 algoV (master) v run linked_list_buggy.v 
==================
/tmp/v/linked_list_buggy.17075566952989225470.tmp.c:1852: error: field not found: link
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```
AFAIK, the "link" word is not reserved in V. But I've found a `c_reserved` array in `v/vlib/v/gen/c/cgen.v`:
```
const (
    // NB: some of the words in c_reserved, are not reserved in C,
    // but are in C++, or have special meaning in V, thus need escaping too.
    c_reserved     = ['auto', .....
'inline', 'int', 'link', 'long', 'malloc', 'namespace',
.....
```
Looks like a collision in the generated C code and my variable name. 
An example of code without "link" word compiles:
https://github.com/ep4sh/algoV/blob/master/linked_list.v

Please let me know if I can help with anything :)

Cheerz,
Pasha

--- a/vlib/v/gen/c/auto_eq_methods.v
+++ b/vlib/v/gen/c/auto_eq_methods.v
@@ -85,30 +85,31 @@ fn (mut g Gen) gen_struct_equality_fn(left_type ast.Type) string {
 				fn_builder.write_string('\n\t\t&& ')
 			}
 			field_type := g.unwrap(field.typ)
+			field_name := c_name(field.name)
 			if field_type.sym.kind == .string {
-				fn_builder.write_string('string__eq(a.$field.name, b.$field.name)')
+				fn_builder.write_string('string__eq(a.$field_name, b.$field_name)')
 			} else if field_type.sym.kind == .sum_type && !field.typ.is_ptr() {
 				eq_fn := g.gen_sumtype_equality_fn(field.typ)
-				fn_builder.write_string('${eq_fn}_sumtype_eq(a.$field.name, b.$field.name)')
+				fn_builder.write_string('${eq_fn}_sumtype_eq(a.$field_name, b.$field_name)')
 			} else if field_type.sym.kind == .struct_ && !field.typ.is_ptr() {
 				eq_fn := g.gen_struct_equality_fn(field.typ)
-				fn_builder.write_string('${eq_fn}_struct_eq(a.$field.name, b.$field.name)')
+				fn_builder.write_string('${eq_fn}_struct_eq(a.$field_name, b.$field_name)')
 			} else if field_type.sym.kind == .array && !field.typ.is_ptr() {
 				eq_fn := g.gen_array_equality_fn(field.typ)
-				fn_builder.write_string('${eq_fn}_arr_eq(a.$field.name, b.$field.name)')
+				fn_builder.write_string('${eq_fn}_arr_eq(a.$field_name, b.$field_name)')
 			} else if field_type.sym.kind == .array_fixed && !field.typ.is_ptr() {
 				eq_fn := g.gen_fixed_array_equality_fn(field.typ)
-				fn_builder.write_string('${eq_fn}_arr_eq(a.$field.name, b.$field.name)')
+				fn_builder.write_string('${eq_fn}_arr_eq(a.$field_name, b.$field_name)')
 			} else if field_type.sym.kind == .map && !field.typ.is_ptr() {
 				eq_fn := g.gen_map_equality_fn(field.typ)
-				fn_builder.write_string('${eq_fn}_map_eq(a.$field.name, b.$field.name)')
+				fn_builder.write_string('${eq_fn}_map_eq(a.$field_name, b.$field_name)')
 			} else if field_type.sym.kind == .alias && !field.typ.is_ptr() {
 				eq_fn := g.gen_alias_equality_fn(field.typ)
-				fn_builder.write_string('${eq_fn}_alias_eq(a.$field.name, b.$field.name)')
+				fn_builder.write_string('${eq_fn}_alias_eq(a.$field_name, b.$field_name)')
 			} else if field_type.sym.kind == .function {
-				fn_builder.write_string('*((voidptr*)(a.$field.name)) == *((voidptr*)(b.$field.name))')
+				fn_builder.write_string('*((voidptr*)(a.$field_name)) == *((voidptr*)(b.$field_name))')
 			} else {
-				fn_builder.write_string('a.$field.name == b.$field.name')
+				fn_builder.write_string('a.$field_name == b.$field_name')
 			}
 		}
 	} else {

--- a/vlib/v/tests/reserved_keywords_as_struct_field_test.v
+++ b/vlib/v/tests/reserved_keywords_as_struct_field_test.v
@@ -1,0 +1,58 @@
+struct Empty {}
+
+// LLNode is struct which holds data and links
+struct LLNode {
+	data int
+	link LinkedList
+}
+
+// LinkedList represent a linked list
+type LinkedList = Empty | LLNode
+
+// insert performs inserting of the value into the LinkedList
+fn insert(ll LinkedList, val int) LinkedList {
+	match ll {
+		Empty {
+			return LLNode{val, Empty{}}
+		}
+		LLNode {
+			return LLNode{
+				...ll
+				link: insert(ll.link, val)
+			}
+		}
+	}
+}
+
+// prepend performs inserting of the value on the top of the LinkedList
+fn prepend(ll LinkedList, val int) LinkedList {
+	match ll {
+		Empty {
+			return LLNode{val, Empty{}}
+		}
+		LLNode {
+			return LLNode{
+				data: val
+				link: ll
+			}
+		}
+	}
+}
+
+fn test_reserved_keywords_as_struct_field() {
+	mut ll := LinkedList(Empty{})
+	ll = insert(ll, 997)
+	ll = insert(ll, 998)
+	ll = insert(ll, 999)
+	mut desired_ll := LinkedList(LLNode{
+		data: 997
+		link: LinkedList(LLNode{
+			data: 998
+			link: LinkedList(LLNode{
+				data: 999
+				link: Empty{}
+			})
+		})
+	})
+	assert ll == desired_ll
+}

